pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go

```diff
@@ -56,10 +58,36 @@ func (pl *TaintToleration) Name() string {
 // failed by this plugin schedulable.
 func (pl *TaintToleration) EventsToRegister() []framework.ClusterEventWithHint {
 	return []framework.ClusterEventWithHint{
-		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterNodeChange},
 	}
 }
 
+// isSchedulableAfterNodeChange is invoked for all node events reported by
+// an informer. It checks whether that change made a previously unschedulable
+// pod schedulable.
+func (pl *TaintToleration) isSchedulableAfterNodeChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	originalNode, modifiedNode, err := util.As[*v1.Node](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterNodeChange", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+
+	wasUntolerated := true
+	if originalNode != nil {
+		_, wasUntolerated = v1helper.FindMatchingUntoleratedTaint(originalNode.Spec.Taints, pod.Spec.Tolerations, helper.DoNotScheduleTaintsFilterFunc())
+	}
+
+	_, isUntolerated := v1helper.FindMatchingUntoleratedTaint(modifiedNode.Spec.Taints, pod.Spec.Tolerations, helper.DoNotScheduleTaintsFilterFunc())
+
+	if wasUntolerated && !isUntolerated {
+		logger.V(4).Info("node was created or updated, and pod tolerate node taint", "pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
+		return framework.QueueAfterBackoff
```

Comment:
shouldn't this be `QueueImmediately` as we can pass the filter on this node now?🤔️

Reply:
There is a related discussion here, let's wait for the outcome of the discussion
https://github.com/kubernetes/kubernetes/pull/119155/files#r1260207009

Reply:
Let's keep afterbackoff in such cases since the conclusion for now is closer to that way  -- to wait the conclusion may take time and we may have a big semantic change based on the discussion. 
The latest status is here: https://github.com/kubernetes/kubernetes/pull/119517

(We can change it later if we need.)